anasey01/BudgetMyFunds
# Implement Docker fixes and User Registration

#### What does this PR do?
- Implements fix for docker
- Implements User Authentication

#### Description of Task to be completed?
Docker
- Run `docker-compose up -d` to build the docker images
- Confirm that it runs with `docker-compose ps`

User Authentication
- Log into your app's instance on docker by running `docker exec -it php bash` 
- Make sure subsequent commands are run in this bash instance
- To create all the required tables, run `php artisan migrate` 4 tables should be created `users, jobs, failed_jobs, cache`

#### How should this be manually tested?
- update this .env values 
```
CACHE_DRIVER=database
QUEUE_CONNECTION=database
BCRYPT_ROUNDS=12
```
- Setup your mailTrap account and Update your `.env` variables to be able to receive mails

User Registration
- `/api/v1/auth/register`
    - email
    - password

User Login

- api/v1/auth/login
    - email
    - password

- Check the jobs table to be sure you have some jobs in there
- To run those jobs, run `php artisan queue:work --tries=3` in your docker-php-bash instance

Enjoy!!!!

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
- [#171017002](https://www.pivotaltracker.com/story/show/171017002)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A
